src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,127 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        size = 0;
+    }
+
+    private int hash(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return (key.hashCode() & (table.length - 1));
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null && value == null) {
```

Comment:
It's not clear why you would check if both key and value are null and then throw an exception. Typically, the value can be null in a hashmap, and it's only necessary to handle the case where the key is null (which you've handled correctly in your hash function).